### PR TITLE
Transaction Database Cron Sync Support

### LINKED ISSUE #312: Develop script to sync SBN DOIs with DOI Service
## 💡 Description

Script should execute via cron and update DOI Service database with SBN DOIs. These DOIs will have different prefixes (10.26033 and 10.26007), but these may eventually be managed by our service, so might as well start syncing them now. In the meantime, if a user tries to update those DOIs from our account, it will fail anyways.

### PR DESCRIPTION
**Pull request by collinss-jpl**
## 🗒️ Summary
This PR makes a number of modifications to the handling of DOI transactions within both the sqlite3 database and the transaction history on disk to better accommodate scheduled synchronizations with DataCite via cron. 

The DOI service now checks to see if there are any changes between a current transaction and the last available record from the transaction history/database before committing. If there have been no changes, including in the stored DataCite JSON label, the transaction is ignored to avoid clogging the database with redundant transactions:

```
INFO __main__:perform_import_to_database Record for DOI 10.26007/xbsh-x639 (urn:nasa:pds:compil-comet:halebopp::1.0) has not changed, skipping...
INFO __main__:perform_import_to_database Record for DOI 10.26007/yr7t-1378 (DS1-C-IDS-3-RDR-BORRELLY-V1.0) has not changed, skipping...
INFO __main__:perform_import_to_database Record for DOI 10.26007/mwyz-6q32 (DS1-C-MICAS-5-BORRELLY-DEM-V1.0) has not changed, skipping...
INFO __main__:perform_import_to_database Record for DOI 10.26007/9dtm-ym89 (EAR-C-5-DDR-PCC-V1.0) has not changed, skipping...
```

Some updates have also been made to better accommodate DOI records submitted by SBN with their own prefixes.

This PR also address several bugs discovered in the service since the last release, including:
* A fix to allow input spreadsheets to provide a "titles" column, instead of strictly enforcing "title"
* A fix to the semantic version sorting done when attempting to derive the newest LIDVID from a list of identifiers associated to a record

## ⚙️ Test Data and/or Report
Unit tests have been updated to account for changes to transaction classes. Unit tests for transaction classes have been updated to test the new duplicate entry checks.
[tox.log](https://github.com/NASA-PDS/doi-service/files/7953796/tox.log)

## ♻️ Related Issues
Partially resolves #312 (cron script still to be developed and installed)

### CODE REVIEW

**Review by nutjob4life — APPROVED** [review]
👍

File: src/pds_doi_service/core/util/general_util.py [hunk 1]
```diff
@@ -28,6 +29,26 @@
...
+    record_payload : str
+        Text contents of a DOI record to generate the checksum for.
+
+    Returns
+    -------
+    hex_digest : str
+        The hex digest of the md5 checksum.
+
+    """
+    md5 = hashlib.md5()
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/pds_doi_service/core/util/general_util.py:47]: ignore

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/pds_doi_service/core/util/general_util.py:47]: ignore

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/pds_doi_service/core/util/general_util.py:47]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (jordanpadams) [reply src/pds_doi_service/core/util/general_util.py:47]: @sonatype-lift ignore

**Pull request merged**